web-components/src/components/usa-link/index.js

```diff
@@ -0,0 +1,69 @@
+import { LitElement, css, html } from "lit";
+
+/**
+ * An example element.
+ *
+ * @slot - This element has a slot
+ * @csspart
+ */
+export class UsaLink extends LitElement {
+  static get properties() {
+    return {
+      /**
+       * The destination URL for the link
+       */
+      href: { type: String },
+    };
+  }
+
+  hasLinkChild() {
+    const childLink = this.querySelector("a");
+    if (!childLink) return false;
+
+    this.href = childLink.href;
+    this.slottedChildren = childLink;
+    this.shadowRoot.appendChild(this.slottedChildren);
+  }
+
+  constructor() {
+    super();
+    this.href = "";
+  }
+
+  render() {
+    return this.hasLinkChild()
+      ? html`<a class="usa-link" href="${this.href}"
+          >${this.slottedChildren}</a
+        >`
+      : html`<a class="usa-link" href="${this.href}"><slot></slot></a>`;
```

Comment:
We could improve readability and make it easier to update if we separated these templates out. Like in the example here - [Composing templates – Lit](https://lit.dev/docs/components/rendering/#composing-templates)

```js
    linkChildTemplate() {
      return html`<a class="usa-link" href="${this.href}"
                >${this.slottedChildren}</a
              >`
    }
    linkSlotTemplate() {
      return html`<a class="usa-link" href="${this.href}"><slot></slot></a>`
    }

    return this.hasLinkChild()
      ? this.linkChildTemplate()
      : this.linkSlotTemplate()
```

Reply:
I'm personally a fan of this. I took a similar approach with the [card component](https://github.com/uswds/uswds-next/pull/13).